services/user/Tokens/plugin/src/query/token_detail.rs

```diff
@@ -0,0 +1,87 @@
+use crate::bindings::common::plugin::{client, server, types as CommonTypes};
+use crate::errors::ErrorType;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+pub struct Welcome {
+    pub data: Data,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Data {
+    pub token_details: TokenDetails,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TokenDetails {
+    pub id: u32,
+    pub owner_details: OwnerDetails,
+    pub precision: Precision,
+    pub symbol_id: String,
+}
```

Comment:
`token_detail` was only made for the plugins internal use, then added to the WIT function `token_owner` as a freebie. 
Could expand on it but thought it would be best to be minimal in the core functionality and have another one for more external use. 

What if `token_owner` is removed and a query plugin functions are added on demand?

Reply:
Moved to queries interface.

Reply:
> What if token_owner is removed and a query plugin functions are added on demand?

Wouldn't mind this either